lldb/tools/lldb-dap/Handler/LaunchRequestHandler.cpp

```diff
@@ -9,127 +9,66 @@
+  dap.SetConfiguration(arguments.configuration, /*is_attach=*/false);
+  dap.last_launch_request = arguments;
+  dap.stop_at_entry = arguments.stopOnEntry;
 
   PrintWelcomeMessage();
 
   // This is a hack for loading DWARF in .o files on Mac where the .o files
   // in the debug map of the main executable have relative paths which
   // require the lldb-dap binary to have its working directory set to that
   // relative root for the .o files in order to be able to load debug info.
+  const std::string debuggerRoot = dap.configuration.debuggerRoot.value_or("");
   if (!debuggerRoot.empty())
     llvm::sys::fs::set_current_path(debuggerRoot);
 
   // Run any initialize LLDB commands the user specified in the launch.json.
   // This is run before target is created, so commands can't do anything with
   // the targets - preRunCommands are run with the target.
-  if (llvm::Error err = dap.RunInitCommands()) {
-    response["success"] = false;
-    EmplaceSafeString(response, "message", llvm::toString(std::move(err)));
-    dap.SendJSON(llvm::json::Value(std::move(response)));
-    return;
-  }
+  if (llvm::Error err = dap.RunInitCommands())
+    return err;
 
-  SetSourceMapFromArguments(*arguments);
+  dap.ConfigureSourceMaps();
 
   lldb::SBError status;
-  dap.SetTarget(dap.CreateTargetFromArguments(*arguments, status));
-  if (status.Fail()) {
-    response["success"] = llvm::json::Value(false);
-    EmplaceSafeString(response, "message", status.GetCString());
-    dap.SendJSON(llvm::json::Value(std::move(response)));
-    return;
-  }
+  dap.SetTarget(dap.CreateTargetFromArguments(
+      arguments.program.value_or(""), arguments.targetTriple.value_or(""),
+      arguments.platformName.value_or(""), status));
+  if (status.Fail())
+    return llvm::make_error<DAPError>(status.GetCString());
 
   // Run any pre run LLDB commands the user specified in the launch.json
-  if (llvm::Error err = dap.RunPreRunCommands()) {
-    response["success"] = false;
-    EmplaceSafeString(response, "message", llvm::toString(std::move(err)));
-    dap.SendJSON(llvm::json::Value(std::move(response)));
-    return;
-  }
+  if (llvm::Error err = dap.RunPreRunCommands())
+    return err;
 
-  status = LaunchProcess(request);
+  if (llvm::Error err = LaunchProcess(arguments))
+    return err;
 
-  if (status.Fail()) {
-    response["success"] = llvm::json::Value(false);
-    EmplaceSafeString(response, "message", std::string(status.GetCString()));
-  } else {
-    dap.RunPostRunCommands();
-  }
+  dap.RunPostRunCommands();
 
-  dap.SendJSON(llvm::json::Value(std::move(response)));
+  return protocol::LaunchResponseBody();
+}
 
-  if (!status.Fail()) {
+void LaunchRequestHandler::PostRun() const {
+  if (dap.target.GetProcess().IsValid()) {
     if (dap.is_attach)
-      SendProcessEvent(dap, Attach); // this happens when doing runInTerminal
+      // this happens when doing runInTerminal
+      SendProcessEvent(dap, Attach);
```

Comment:
With the comment moved, this (and the else-case) now need braces 🙃

Reply:
Adjusted this to use a ternary if and moved the comment.